langchain/document_loaders/mongodb.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+from typing import List
+
+from motor.motor_asyncio import AsyncIOMotorClient
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class MongodbLoader(BaseLoader):
+    """Load MongoDB documents."""
+
+    def __init__(
+        self,
+        connection_string: str,
+        db_name: str,
+        collection_name: str,
+    ):
+        self.client = AsyncIOMotorClient(connection_string)
+        self.db_name = db_name
+        self.collection_name = collection_name
+        self.db = self.client.get_database(db_name)
+        self.collection = self.db.get_collection(collection_name)
+
+        if not connection_string:
+            raise ValueError("connection_string must be provided.")
+
+        if not db_name:
+            raise ValueError("db_name must be provided.")
+
+        if not collection_name:
+            raise ValueError("collection_name must be provided.")
+
+    def load(self) -> List[Document]:
+        result = asyncio.run(self._async_load())
+        return result
+
+    async def _async_load(self) -> List[Document]:
+        result = []
+
+        try:
+            cursor = self.collection.find()
+
+            async for doc in cursor:
+                content = str(doc)
+                metadata = {
+                    "database": self.db_name,
+                    "collection": self.collection_name,
+                }
+                result.append(Document(page_content=content, metadata=metadata))
+        except Exception as e:
+            print(f"Error: {e}")
```

Comment:
I’m thinking of what value we can get when the loading is partially finished. It might be good to raise exception when any of the loading failed but it depends on the biz needs.

Reply:
@skcoirz I added a check for self.collection.count_documents to verify this matches the total amount returned, and added a test case for this.

Reply:
Note that the PR will still fail because the motor package is not currently part of Langchain poetry file.  Also there's a mypy error on the test file because it doesn't understand some types related to the missing motor package.